vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/loader/LocalJavaProjectLoader.java

```diff
@@ -83,45 +84,49 @@ public Optional<IJavaProject> load(String uri, Map<Event, Consumer<String>> hand
             {
-                ofNullable(handlers.get(Event.CORRUPTED)).ifPresent(c -> c.accept(absProjectPath));
+                onError.accept(String.format("Project file by path '%s' is corrupted", absProjectPath));
                 return Optional.empty();
             }
 
-            ofNullable(handlers.get(Event.LOADED)).ifPresent(c -> c.accept(description.getName()));
+            onLoad.accept(String.format("Project with the name '%s' is loaded", description.getName()));
             return Optional.of(projectTransformer.apply(project));
         }, VividusStudioException::new);
     }
 
     @SuppressWarnings("restriction")
-    public void reload(IProject project, Consumer<String> messageConsumer) throws Exception
+    public void reload(IProject project, Consumer<String> onInfo, Consumer<String> onError) throws Exception
     {
         JavaModelManager.PerProjectInfo perProjectInfo = JavaModelManager.getJavaModelManager()
                 .getPerProjectInfo(project, true);
         perProjectInfo.setRawClasspath(null, null, null);
         perProjectInfo.resetResolvedClasspath();
         perProjectInfo.forgetExternalTimestampsAndIndexes();
 
-        build(project.getLocation().toFile(), Optional.of(messageConsumer));
+        build(project.getLocation().toFile(), onInfo, onError);
         project.refreshLocal(IResource.DEPTH_INFINITE, null);
     }
 
-    private static void build(File projectFolder, Optional<Consumer<String>> messageConsumer) throws Exception
+    private static void build(File projectFolder, Consumer<String> onInfo, Consumer<String> onError) throws Exception
     {
         BuildConfiguration config = BuildConfiguration.forRootProjectDirectory(projectFolder)
                                                       .build();
 
         GradleBuild build = GradleCore.getWorkspace().createBuild(config);
 
-        build.withConnection(connection ->
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream())
         {
-            BuildLauncher launcher = connection.newBuild().forTasks("eclipse");
-            messageConsumer.ifPresent(msgConsumer ->
+            build.withConnection(connection ->
             {
-                ProgressListener listener = event -> msgConsumer.accept(event.getDescription());
+                BuildLauncher launcher = connection.newBuild().setStandardOutput(outputStream).forTasks("eclipse");
+                ProgressListener listener = event -> onInfo.accept(event.getDescription());
                 launcher.addProgressListener(listener);
-            });
-            launcher.run();
-            return null;
-        }, new NullProgressMonitor());
+                launcher.run();
+                return null;
+            }, new NullProgressMonitor());
+
+            String output = new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
+            Splitter.split(output).stream().filter(l -> l.startsWith("Could not resolve:"))
```

Comment:
relying on text output is not the best approach, however I don't see other integration points